os/fs/smartfs/smartfs_smart.c

```diff
@@ -693,7 +753,66 @@ static int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_of
 	}
 #endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+	/* File's data sector has been synced with MTD, now check if this is a new file entry and write the entry */
+	if (sf->bflags && SMARTFS_BFLAG_NEW_ENTRY) {
```

Comment:
please check how sf->bflags is changed.
I think only first time it include SMARTFS_BFLAG_NEWALLOC after second time, it doesn't have it.
Because after sync sf->bflags is asigned as SMARTFS_BFLAG_DIRTY.
so below will be work. please check it.
```
if (sf->bflags & SMARTFS_BFLAG_NEWALLOC) {
     write_first_data_sector;
      write_entry;
}

Reply:
bflags |= NEW_ENTRY will be done only when a new entry is created for the file.
This will be done in smartfs_open

During sync, this entry will be written to MTD,
After that, checking for new entry is no longer required.